json-endpoints/class.wpcom-json-api-update-user-endpoint.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+class WPCOM_JSON_API_Update_User_Endpoint extends WPCOM_JSON_API_Endpoint {
+
+	function callback( $path = '', $blog_id = 0, $user_id = 0 ) {
+		$blog_id = $this->api->switch_to_blog_and_validate_user( $this->api->get_blog_id( $blog_id ) );
+		if ( is_wp_error( $blog_id ) ) {
+			return $blog_id;
+		}
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			if ( wpcom_get_blog_owner( $blog_id ) == $user_id ) {
+				return new WP_Error( 'forbidden', 'A site owner can not be removed through this endpoint.', 403 );
+			}
+		}
+
+		if ( $this->api->ends_with( $path, '/delete' ) ) {
+			return $this->delete_or_remove_user( $user_id );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if a user exists by checking to see if a WP_User object exists for a user ID.
+	 * @param  int $user_id
+	 * @return bool
+	 */
+	function user_exists( $user_id ) {
+		$user = get_user_by( 'id', $user_id );
+
+		return false != $user && is_a( $user, 'WP_User' );
+	}
+
+	/**
+	 * Validates user input and then decides whether to remove or delete a user.
+	 * @param  int $user_id
+	 * @return array|WP_Error
+	 */
+	function delete_or_remove_user( $user_id ) {
+		if ( 0 == $user_id ) {
+			return new WP_Error( 'invalid_input', 'A valid user ID must be specified.', 400 );
+		}
+
+		if ( get_current_user_id() == $user_id ) {
+			return new WP_Error( 'invalid_input', 'User can not remove or delete self through this endpoint.', 400 );
+		}
```

Comment:
In a prior conversation I mentioned that you should add a check to ensure that you are not removing the only admin for a site. The above conditional should handle that, so I think you are all good!